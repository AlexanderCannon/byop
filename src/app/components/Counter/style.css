.byop-app__counter {
  align-content: center;
  justify-content: center;
  display: flex;
  flex-direction: column;
  min-width: 300px;
  color: var(--color-white);
  text-shadow: var(--text-shadow)
}
