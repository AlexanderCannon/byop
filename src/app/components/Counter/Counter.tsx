import Button from "@material-ui/core/Button";
import Typography from "@material-ui/core/Typography";
import { CSSProperties } from "@material-ui/styles";
import * as React from "react";
import "./style.css";

interface IProps {
  initialCount?: number;
}

const style: CSSProperties = { backgroundColor: "blue" };

const Counter = ({ initialCount = 0 }: IProps) => {
  const [count, setCount] = React.useState(initialCount);
  React.useEffect(() => {
    document.title = `You clicked ${count} times`;
  });

  const increment = () => setCount(count + 1);

  return (
    <div className="byop-app--counter" style={style}>
      <Typography variant="h3">
        You clicked{" "}
        <span data-testid="byop-app--counter-count-value">{count}</span> times
      </Typography>

      <Button
        className="byop--app-counter-button"
        data-testid="byop-app--counter-button"
        variant="contained"
        color="primary"
        onClick={increment}>
        Click me
      </Button>
    </div>
  );
};

export default Counter;
