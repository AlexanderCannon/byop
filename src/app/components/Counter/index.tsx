import Button from "@material-ui/core/Button";
import Typography from "@material-ui/core/Typography";
import * as React from "react";
import "./style.css";

interface IProps {
  initialCount?: number;
}

interface IState {
  count: number;
}

const increment = Symbol();
export default class Counter extends React.Component<IProps, IState> {
  public static defaultProps = {
    initialCount: 0
  };
  constructor(props: IProps) {
    super(props);
    this.state = {
      count: props.initialCount!
    };
    this[increment] = this[increment].bind(this);
  }
  public [increment] = () => {
    const { count } = this.state;
    this.setState({ count: count + 1 });
  };
  public render() {
    const { count } = this.state;
    document.title = `You clicked ${count} times`;
    return (
      <div className="byop-app--counter">
        <Typography variant="h3">
          You clicked{" "}
          <span data-testid="byop-app--counter-count-value">{count}</span> times
        </Typography>
        <Button
          data-testid="byop-app--counter-button"
          variant="contained"
          color="primary"
          onClick={this[increment]}>
          Click me
        </Button>
      </div>
    );
  }
}
