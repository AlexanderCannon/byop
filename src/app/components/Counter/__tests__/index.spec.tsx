import Button from "@material-ui/core/Button";
import { fireEvent, getByTestId, render } from "@testing-library/react";
import { shallow } from "enzyme";
import * as React from "react";
import reactTestRenderer from "react-test-renderer";
// tslint:disable-next-line: import-name
import Counter from "../index";

/*
 * Tests using react-test-renderer
 */

test("Component renders", () => {
  const counter = reactTestRenderer.create(<Counter />);
  const tree = counter.toJSON();
  expect(tree).toMatchSnapshot();
});

test("Component renders with an initial count", () => {
  const counter = reactTestRenderer.create(<Counter initialCount={2} />);
  const tree = counter.toJSON();
  expect(tree).toMatchSnapshot();
});

/*
 * Tests using enzyme
 */

test("should contain a button", () => {
  const counter = shallow(<Counter />);
  expect(counter.find(Button)).toHaveLength(1);
});
test("Component increments count when button is clicked", () => {
  const counter = shallow(<Counter initialCount={0} />);
  expect(counter.text()).toEqual("You clicked 0 timesClick me");
  counter.find(Button).simulate("click");
  expect(counter.text()).toEqual("You clicked 1 timesClick me");
  counter.find(Button).simulate("click");
  counter.find(Button).simulate("click");
  expect(counter.text()).toEqual("You clicked 3 timesClick me");
});

/*
 * Tests using react-testing-library
 */
test("App loads with initial state of 0", () => {
  const { container } = render(<Counter />);
  const countValue = getByTestId(container, "byop-app__counter-count-value");
  expect(countValue.textContent).toBe("0");
});

test("That the button works", () => {
  const { container } = render(<Counter />);
  const countValue = getByTestId(container, "byop-app__counter-count-value");
  const increment = getByTestId(container, "byop-app__counter-button");
  expect(countValue.textContent).toBe("0");
  fireEvent.click(increment);
  expect(countValue.textContent).toBe("1");
  fireEvent.click(increment);
  fireEvent.click(increment);
  expect(countValue.textContent).toBe("3");
});
