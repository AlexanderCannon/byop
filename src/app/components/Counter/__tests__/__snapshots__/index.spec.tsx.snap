// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component renders 1`] = `
<div
  className="byop-app__counter"
>
  <h3
    className="MuiTypography-root MuiTypography-h3"
  >
    You clicked
     
    <span
      data-testid="byop-app__counter-count-value"
    >
      0
    </span>
     times
  </h3>
  <button
    className="MuiButtonBase-root MuiButton-root byop-app__counter-button MuiButton-contained MuiButton-containedPrimary"
    data-testid="byop-app__counter-button"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      Click me
    </span>
  </button>
</div>
`;

exports[`Component renders with an initial count 1`] = `
<div
  className="byop-app__counter"
>
  <h3
    className="MuiTypography-root MuiTypography-h3"
  >
    You clicked
     
    <span
      data-testid="byop-app__counter-count-value"
    >
      2
    </span>
     times
  </h3>
  <button
    className="MuiButtonBase-root MuiButton-root byop-app__counter-button MuiButton-contained MuiButton-containedPrimary"
    data-testid="byop-app__counter-button"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      Click me
    </span>
  </button>
</div>
`;
